src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  hasLoader: boolean;
+  onDeleteTodo: (id: number) => void;
+  onToggleTodo: (id: number, completed: boolean) => void;
+  onChangeTodo: (id: number, title: string) => Promise<void>;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  hasLoader,
+  onDeleteTodo,
+  onToggleTodo,
+  onChangeTodo,
+}) => {
+  const { id, title, completed } = todo;
+  const [isEditingTodo, setIsEditingTodo] = useState<boolean>(false);
+  const editingTodoField = useRef<HTMLInputElement | null>(null);
+
+  const changeTitle = async (idTodo: number, titleTodo: string) => {
+    try {
+      await onChangeTodo(idTodo, titleTodo);
+      setIsEditingTodo(false);
+    } catch (e) {
+      editingTodoField.current?.focus();
```

Comment:
why focus on error? maybe it's better to show some error, WDYT?

Reply:
Fixed it.  I've made as in original